internal/primitive/transform/function/strings_functions.go

```diff
@@ -118,3 +118,28 @@ var CapitalizeSentence = function{
 		return strings.ToUpper(string(value[0])) + value[1:], nil
 	},
 }
+
+var ReplaceBetweenDelimitersFunction = function{
+	name:		"REPLACE_BETWEEN_DELIMITERS",
+	fixedArgs:	[]common.Type{common.String, common.String, common.String, common.String},
+	fn:	func(args []interface{}) (interface{}, error) {
+		path, _ := args[0].(string)
+		startPattern := args[1].(string)
+		endPattern := args[2].(string)
+		newValue := args[3].(string)
+
+		if (startPattern != endPattern && strings.Contains(path, startPattern) && strings.Contains(path, endPattern)) {
+			firstSplit := strings.Split(path, startPattern)
+			secondSplit := strings.Split(firstSplit[1], endPattern)
+			secondSplit[0] = startPattern + newValue + endPattern
+			return firstSplit[0] + secondSplit[0] + secondSplit[1], nil
+		} else if startPattern == endPattern && strings.Contains(path, startPattern) {
+			firstSplit := strings.Split(path, startPattern)
+			firstSplit[1] = startPattern + newValue + endPattern
+			return firstSplit[0] + firstSplit[1] + firstSplit[2], nil
```

Comment:
it will panic if the original string only has one delimiter,  such as `Hello, &&World, end`, the code
```
firstSplit := strings.Split(path, startPattern)
```
len(firstSplit) is 2

Reply:
Please for clarity sake, should I write the code to replace the rest of the string where the only delimiter start if it's one delimiter present or return error?

Reply:
I think it needs to return an error